Gems/ROS2PoseControl/Code/Source/Clients/ROS2PoseControl.cpp

```diff
@@ -75,19 +75,21 @@ namespace ROS2PoseControl
     AZ::Outcome<AZ::Transform, const char*> ROS2PoseControl::GetCurrentTransformViaTF2() const
     {
         geometry_msgs::msg::TransformStamped transformStamped;
-        if (m_tf_buffer->canTransform(m_configuration.m_referenceFrame.c_str(), m_configuration.m_targetFrame.c_str(), tf2::TimePointZero))
-        {
+        std::string errorString;
+        if (m_tf_buffer->canTransform(m_configuration.m_referenceFrame.c_str(), m_configuration.m_targetFrame.c_str(),
+                                      tf2::TimePointZero, &errorString)) {
             transformStamped = m_tf_buffer->lookupTransform(
                 m_configuration.m_referenceFrame.c_str(), m_configuration.m_targetFrame.c_str(), tf2::TimePointZero);
         }
         else
         {
-            AZ_Warning(
+            AZ_WarningOnce(
```

Comment:
This is too much of a shortcut. We should also know if the tf will disappear at some point in time.

Consider setting timeout for `canTransform` for non-zero and leave this warning to default implementation (not "Once")

Reply:
I added a flag that will make it so warring happens only once. I'm not sure what treatment is needed for other warring related to ground, as this feature can be toggled.